toolchain/check/testdata/return/import_convert_function.carbon

```diff
@@ -1664,96 +1703,91 @@ fn F0(n: i32) -> P.D {
 // CHECK:STDOUT:   %Make.ref: %Make.type = name_ref Make, imports.%P.Make [concrete = constants.%Make]
 // CHECK:STDOUT:   %.loc6_15: ref %D = splice_block %return {}
 // CHECK:STDOUT:   %Make.call: init %D = call %Make.ref() to %.loc6_15
-// CHECK:STDOUT:   %T.as.Destroy.impl.Op.bound.loc6: <bound method> = bound_method %.loc6_15, constants.%T.as.Destroy.impl.Op.f5a
-// CHECK:STDOUT:   %T.as.Destroy.impl.Op.specific_fn.73: <specific function> = specific_function constants.%T.as.Destroy.impl.Op.f5a, @T.as.Destroy.impl.Op(constants.%D) [concrete = constants.%T.as.Destroy.impl.Op.specific_fn.bd1]
-// CHECK:STDOUT:   %bound_method.loc6: <bound method> = bound_method %.loc6_15, %T.as.Destroy.impl.Op.specific_fn.73
+// CHECK:STDOUT:   %D.as.Destroy.impl.Op.bound.loc6: <bound method> = bound_method %.loc6_15, constants.%D.as.Destroy.impl.Op
```

Comment:
This is a pre-existing problem, but I just wanted to make sure it's on your radar: the SemIR for this function seems to be growing quadratically, because every return is generating destructor calls for all the prior temporaries. I see two issues with that:
- The temporaries for the prior branches aren't actually live, so it's incorrect as well as inefficient to destroy them.
- Even if we fix that, we'll still get quadratic behavior if there genuinely are a linear number of live variables at each return. I think you've alluded to eventually emitting destructor calls in reusable blocks instead of in-line, which could fix this problem.